changelog/6003.feature.rst

```diff
@@ -0,0 +1 @@
+Improve short excinfo with LineMatcher failures, via new ``OutcomeException.short_msg``.
```

Comment:
```suggestion
Improve short excinfo with LineMatcher failures in short test summaries, via new ``OutcomeException.short_msg``.
```
?

Reply:
The change is good, but I also suggest to rename it to `6003.improvement.rst`. 👍